http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClientBuilder.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.client.netty;
+
+import io.micronaut.core.annotation.AnnotationMetadataResolver;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.core.util.ArgumentUtils;
+import io.micronaut.http.bind.RequestBinderRegistry;
+import io.micronaut.http.body.MessageBodyHandlerRegistry;
+import io.micronaut.http.client.HttpClientConfiguration;
+import io.micronaut.http.client.HttpVersionSelection;
+import io.micronaut.http.client.LoadBalancer;
+import io.micronaut.http.client.filter.ClientFilterResolutionContext;
+import io.micronaut.http.client.filter.DefaultHttpClientFilterResolver;
+import io.micronaut.http.client.netty.ssl.ClientSslBuilder;
+import io.micronaut.http.codec.MediaTypeCodecRegistry;
+import io.micronaut.http.filter.HttpClientFilter;
+import io.micronaut.http.filter.HttpClientFilterResolver;
+import io.micronaut.http.filter.HttpFilterResolver;
+import io.micronaut.websocket.context.WebSocketBeanRegistry;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFactory;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.socket.nio.NioDatagramChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.resolver.AddressResolverGroup;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ThreadFactory;
+
+/**
+ * While {@link DefaultHttpClient} is internal API, there are a few uses outside micronaut-core
+ * that use it directly, in particular micronaut-oracle-cloud. This builder acts as API for those
+ * users.
+ * <p>
+ * If you need to make a method of this builder public, please document the module that uses it.
+ *
+ * @author Jonas Konrad
+ * @since 4.7.0
+ */
+@Internal
+public final class DefaultHttpClientBuilder {
+    @Nullable
+    LoadBalancer loadBalancer = null;
```

Comment:
can all these fields be made `private`?

Reply:
No they are all accessed from DefaultHttpClient